### PR TITLE
Feature/ab2d 2640 handle multiple mbis

### PR DESCRIPTION
**Pull request by wnyffenegger**
**JIRA Tickets:**

[AB2D-2640](https://jira.cms.gov/browse/AB2D-2640) - Handle Multiple MBIs
 
### What Does This PR Do?

Major Changes:
* Reactivates MBI functionality which was temporarily disabled.
* Remove crosswalk endpoint.
* Differentiate between current and historic mbis when ingesting in PatientContractCallable.
* Write out mbis with annotations for whether the MBIs are current or historic.
* Tests for the changes

Classes Changed:
* processor.PatientContractCallable
* processor.ConctractProcessorImpl
* process.domainmodel.Identifiers
* e2e-test.TestRunner

### What Should Reviewers Watch For?

* Changes to PatientContractCallable which may cause exceptions if expected fields are null or missing
* Changes to ContractProcessorImpl which may increase the size of export files
* Changes to ContractProcessorImpl which may cause exceptions in edge cases where no mbis are found.
* That the tests catch all obvious edge cases

### Security Implications

- [ ] This PR adds new software dependencies
- [ ] This PR modifies or invalidates our security controls
- [ ] This PR stores or transmits data that was not stored or transmitted before

- [ ] This PR requires additional review of its security implications for other reasons

### Submitter Checklist

- [ ] This PR includes any required documentation changes, (`README`, changelog / release notes, website).
- [ ] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments.
- [ ] Code checked for PHI/PII exposure

### CODE REVIEW

File: worker/src/test/java/gov/cms/ab2d/worker/processor/BundleUtils.java [hunk 1]
```diff
@@ -33,9 +45,27 @@ public static Bundle createBundle(Bundle.BundleEntryComponent ... bundleEntries)
...
+        return createPatient(patientId, mbi, year, true);
+    }
+
+    public static Patient createPatientWithMultipleMbis(String patientId, int numMbis, int year) {
+        var patient = new Patient();
+        patient.getIdentifier().add(createBeneficiaryIdentifier(patientId));
+
+        patient.getIdentifier().add(createMbiIdentifier(patientId + "mbi-" + 0, true));
+
+        for (int i = 1; i < numMbis; i++) {
```
Comment (nflearl) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/BundleUtils.java:57]: nitpick
          i => idx

It's a whole lot easier to search for 3 letter loop indexes rather than 1.

**Review by nflearl — CHANGES_REQUESTED** [review]
Found one small nit to fix.

**Review by lailasharshar — APPROVED** [review]

**Review by lailasharshar — DISMISSED** [review]

File: e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java [hunk 2]
```diff
@@ -64,6 +65,9 @@
 
     private static final String FHIR_TYPE = "application/fhir+ndjson";
 
+    private static final String CURRENCY_IDENTIFIER =
```
Comment (nflearl) [inline e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java:68]: This is a duplicate definition.  Please consolidate into using only one.

File: worker/src/test/java/gov/cms/ab2d/worker/processor/BundleUtils.java [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
 
 public class BundleUtils {
 
     public static final String BENEFICIARY_ID = "https://bluebutton.cms.gov/resources/variables/bene_id";
     public static final String MBI_ID = "http://hl7.org/fhir/sid/us-mbi";
 
+    public static final String CURRENCY_IDENTIFIER =
+            "https://bluebutton.cms.gov/resources/codesystem/identifier-currency";
+    public static final String CURRENT = "current";
+    public static final String HISTORIC = "historic";
```
Comment (nflearl) [inline worker/src/test/java/gov/cms/ab2d/worker/processor/BundleUtils.java:18]: Rename to something like:
CURRENT_MBI
HISTORIC_MBI

to distinguish.

CURRENT also conflicts with what is in Enumerations.java.  Perhaps those definitions can be of use.

File: worker/src/main/java/gov/cms/ab2d/worker/processor/PatientContractCallable.java [hunk 4]
```diff
@@ -5,20 +5,26 @@
...
+import static java.util.stream.Collectors.*;
 
 @Slf4j
 public class PatientContractCallable implements Callable<ContractMapping> {
     private static final String BENEFICIARY_ID = "https://bluebutton.cms.gov/resources/variables/bene_id";
+    static final String CURRENCY_IDENTIFIER =
+            "https://bluebutton.cms.gov/resources/codesystem/identifier-currency";
+    static final String CURRENT = "current";
+    static final String HISTORIC = "historic";
+
```
Comment (nflearl) [inline worker/src/main/java/gov/cms/ab2d/worker/processor/PatientContractCallable.java:25]: Another duplication and need for renaming CURRENT and HISTORIC>

**Review by nflearl — CHANGES_REQUESTED** [review]
Consolidate and rename constants.

File: e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java (hunk 2 above)
Comment (wnyffenegger) [reply e2e-test/src/test/java/gov/cms/ab2d/e2etest/TestRunner.java:68]: Can't dedup this one because it isn't a shared dependency of the worker and I don't want to move it to the common module.

**Review by nflearl — APPROVED** [review]

**Review by lailasharshar — APPROVED** [review]
